src/fidesops/api/v1/endpoints/saas_config_endpoints.py

```diff
@@ -0,0 +1,168 @@
+)
+def validate_saas_config(
+    saas_config: SaaSConfig,
+) -> ValidateSaaSConfigResponse:
+    """
+    Uses the SaaSConfig Pydantic model to validate the SaaS config
+    without attempting to save it to the database.
+
+    Checks that:
+    - all required fields are present, all field values are valid types
+    - each connector_param only has one of references or identity, not both
+    """
+
+    logger.info(f"Validation successful for SaaS config '{saas_config.fides_key}'")
+    return ValidateSaaSConfigResponse(
+        saas_config=saas_config,
+        validation_details=SaaSConfigValidationDetails(
+            msg="Validation successful",
+        ),
+    )
+
+
+@router.patch(
+    SAAS_CONFIG,
+    dependencies=[Security(verify_oauth_client, scopes=[SAAS_CONFIG_CREATE_OR_UPDATE])],
+    status_code=200,
+    response_model=SaaSConfig,
+)
+def patch_saas_config(
+    saas_config: SaaSConfig,
+    db: Session = Depends(deps.get_db),
+    connection_config: ConnectionConfig = Depends(_get_saas_connection_config),
+) -> SaaSConfig:
+    """
+    Given a SaaS config element, update the corresponding ConnectionConfig object
+    or report failure
+    """
+    logger.info(
+        f"Updating SaaS config '{saas_config.fides_key}' on connection config '{connection_config.key}'"
+    )
+    connection_config.update(db, data={"saas_config": saas_config.dict()})
+    return connection_config.saas_config
+
+
+@router.get(
+    SAAS_CONFIG,
+    dependencies=[Security(verify_oauth_client, scopes=[SAAS_CONFIG_READ])],
+    response_model=SaaSConfig,
+)
+def get_saas_config(
+    connection_config: ConnectionConfig = Depends(_get_saas_connection_config),
+) -> SaaSConfig:
+    """Returns the SaaS config for the given connection config."""
+
+    logger.info(f"Finding SaaS config for connection '{connection_config.key}'")
+    saas_config = connection_config.saas_config
+    if not saas_config:
+        raise HTTPException(
+            status_code=HTTP_404_NOT_FOUND,
+            detail=f"No SaaS config found for connection '{connection_config.key}'",
+        )
+    return connection_config.saas_config
+
+
+@router.delete(
+    SAAS_CONFIG,
+    dependencies=[Security(verify_oauth_client, scopes=[SAAS_CONFIG_DELETE])],
+    status_code=204,
+)
+def delete_saas_config(
+    db: Session = Depends(deps.get_db),
+    connection_config: ConnectionConfig = Depends(_get_saas_connection_config),
+) -> None:
+    """Removes the SaaS config for the given connection config.
+    The corresponding dataset and secrets must be deleted before deleting the SaaS config"""
+
+    logger.info(f"Finding SaaS config for connection '{connection_config.key}'")
+    saas_config = connection_config.saas_config
+    if not saas_config:
+        raise HTTPException(
+            status_code=HTTP_404_NOT_FOUND,
+            detail=f"No SaaS config found for connection '{connection_config.key}'",
+        )
+
+    fides_key = saas_config["fides_key"]
```

Comment:
I know we're using schemas to validate this data on entry, but since this is a retrieval it could be susceptible to dirty data. It's worth considering `.get("fides_key", <default>)` here

Reply:
Using `.get("fides_key")` and throwing an early exception if the value is None.